Orange/widgets/gui/_checkbox.py

```diff
@@ -0,0 +1,68 @@
+from AnyQt import QtWidgets
+from AnyQt.QtCore import Qt
+
+from Orange.widgets.gui.base import miscellanea, connectControl, Disabler, \
+    FunctionCallback
+from Orange.widgets.utils import getdeepattr
+
+from ._boxes import hBox
+
+def checkBox(widget, master, value, label, box=None,
+             callback=None, getwidget=False, id_=None, labelWidth=None,
+             disables=None, **misc):
+    """
+    A simple checkbox.
+
+    :param widget: the widget into which the box is inserted
+    :type widget: QWidget or None
+    :param master: master widget
+    :type master: OWWidget or OWComponent
+    :param value: the master's attribute with which the value is synchronized
+    :type value:  str
+    :param label: label
+    :type label: str
+    :param box: tells whether the widget has a border, and its label
+    :type box: int or str or None
+    :param callback: a function that is called when the check box state is
+        changed
+    :type callback: function
+    :param getwidget: If set `True`, the callback function will get a keyword
+        argument `widget` referencing the check box
+    :type getwidget: bool
+    :param id_: If present, the callback function will get a keyword argument
+        `id` with this value
+    :type id_: any
+    :param labelWidth: the width of the label
+    :type labelWidth: int
+    :param disables: a list of widgets that are disabled if the check box is
+        unchecked
+    :type disables: list or QWidget or None
+    :return: constructed check box; if is is placed within a box, the box is
+        return in the attribute `box`
+    :rtype: QCheckBox
+    """
+    if box:
+        b = hBox(widget, box, addToLayout=False)
+    else:
+        b = widget
+    cbox = QtWidgets.QCheckBox(label, b)
+
+    if labelWidth:
+        cbox.setFixedSize(labelWidth, cbox.sizeHint().height())
+    cbox.setChecked(getdeepattr(master, value))
+
+    values = [Qt.Unchecked, Qt.Checked, Qt.PartiallyChecked]
+    connectControl(
+        master, value, None, cbox.toggled[bool],
+        cfront=lambda val: val is not None and cbox.setCheckState(values[val]),
```

Comment:
Callfront classes were needed just to store all needed variables. Closures are a better place for this. Callfront classes will thus be replaced with simple lambdas like this one here (or with functions, where necessary).

(Another reason for call front classes, besides storing variables, was that they prevented cyclic callbacks. `connectControl` will take care of this by wrapping these lambdas into classes, but we don't need to know about that here -- this belongs to `connectControl`.)